### PR TITLE
MULE-9519: Refactor extensions packaging to be compatible with new pl…

### PR DESCRIPTION
**Pull request by pablomusumeci**
…ugin architecture

### CODE REVIEW

**Comment by asequeira** [comment]
This shouldn't be merged until the mule-extensions-maven-plugin PR is merged and an snapshot is built

File: extensions/file/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
     <groupId>org.mule.modules</groupId>
     <artifactId>mule-module-file</artifactId>
-    <packaging>jar</packaging>
+    <packaging>mule-module</packaging>
```
Comment (marianogonzalez) [inline extensions/file/pom.xml]: this should be mule-extension

File: extensions/pom.xml [hunk 2]
```diff
@@ -98,6 +102,12 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (marianogonzalez) [inline extensions/pom.xml]: these changes mean that extensions/all-extensions/pom.xml can no longer have this pom as a parent.

File: extensions/all-extensions/pom.xml [hunk 3]
```diff
@@ -2,8 +2,8 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
-        <groupId>org.mule.extensions</groupId>
-        <artifactId>mule-extensions</artifactId>
+        <groupId>org.mule</groupId>
+        <artifactId>mule</artifactId>
```
Comment (asequeira) [inline extensions/all-extensions/pom.xml]: This groupId:artifactId already exists, my understanding is that it should be org.mule.extensions.mule-extensions

**Pull request merged**